paasta_tools/cli/cmds/spark_run.py

```diff
@@ -602,6 +613,37 @@ def run_docker_container(
     return 0
 
 
+def get_spark_app_name(
+    original_docker_cmd: Union[Any, str, List[str]], spark_ui_port: int
+) -> str:
+    # Use submitted batch name as default spark_run job name
+    spark_app_name = "paasta_spark_run"
+
+    docker_cmds = (
+        shlex.split(original_docker_cmd)
+        if isinstance(original_docker_cmd, str)
+        else original_docker_cmd
+    )
+
+    if "spark-submit" in original_docker_cmd:
```

Comment:
Looks like commands like "python s3.py" also benefit from the change. Not many such Spark jobs though.

Reply:
yup, that would work, too!